File: epi_judge_python/string_integer_interconversion.py
from test_framework import generic_test
from test_framework.test_failure import TestFailure
import functools

def int_to_string(x):
    # TODO - you fill in here.
    neg = False
    if x < 0:
        x = -x
        neg = True
    res = ""
    while True:
        res = chr(ord('0') + (x % 10)) + res
        x = x // 10
        if x == 0:
            break
    if neg:
        res = '-' + res
    return res


def string_to_int(s):
    # TODO - you fill in here.
    # res = 0
    # neg = False
    # if s[0] == '-':
    #     neg = True
    #     s = s[1:]
    # for i,c in enumerate(reversed(s)):
    #     res += (ord(c) - ord('0')) * 10 ** i
    # if neg:
    #     res *= -1
    res = functools.reduce(lambda acc,x: acc * 10 + (ord(x)-ord('0')) ,s[(s[0]=='-'):],0)
    return res * (-1 if s[0] == '-' else 1)


def wrapper(x, s):
    if int_to_string(x) != s:
        raise TestFailure("Int to string conversion failed")
    if string_to_int(s) != x:
        raise TestFailure("String to int conversion failed")


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("string_integer_interconversion.py",
                                       'string_integer_interconversion.tsv',
                                       wrapper))

File: epi_judge_java/epi/StringIntegerInterconversion.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
import epi.test_framework.TestFailure;

public class StringIntegerInterconversion {

  public static String intToString(int x) {
    if (x == 0){
      return "0";
    }
    StringBuilder res = new StringBuilder();
    boolean neg = false;
    if (x < 0){
      neg = true;
    }
    while (x != 0){
      char offset = (char) Math.abs((x % 10));
      offset += '0';
      x /= 10;

      res.append(offset);
    }

    if (neg){
      res.append('-');
    }

    String result = res.reverse().toString();
    return result;
  }
  public static int stringToInt(String s) {
    int res = 0;
    boolean negative = false;

    if (s.charAt(0) == '-'){
      negative = true;
      s = s.substring(1,s.length());
    }
    for (int i = 0; i < s.length(); i ++){
      res *= 10;
      res += s.charAt(i) - '0';
    }

    return negative ? -res : res;
  }
  @EpiTest(testDataFile = "string_integer_interconversion.tsv")
  public static void wrapper(int x, String s) throws TestFailure {
    if (!intToString(x).equals(s)) {
      throw new TestFailure("Int to string conversion failed");
    }
    if (stringToInt(s) != x) {
      throw new TestFailure("String to int conversion failed");
    }
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "StringIntegerInterconversion.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
